keras/layers/pooling/global_average_pooling1d.py

```diff
@@ -82,6 +82,8 @@ def __init__(self, data_format="channels_last", **kwargs):
 
     def call(self, inputs, mask=None):
         steps_axis = 1 if self.data_format == "channels_last" else 2
+        if inputs.shape[steps_axis] == 0:
+            raise ValueError("Reducing axis cannot be of 0 dimension")
```

Comment:
Move this check to `build`, since it's a static input shape check.

Error message should be: `"Incorrect input shape {input_shape} with dimension 0 at reduction axis {axis}."`